module/trace/constants.go

```diff
@@ -8,33 +8,46 @@ const (
 
 	// State
 	// mutator.Extend - full payload check
-	ProtoStateMutatorExtend                = "common.state.proto.mutator.extend"
-	ProtoStateMutatorExtendCheckHeader     = "common.state.proto.mutator.extend.checkHeader"
-	ProtoStateMutatorExtendCheckGuarantees = "common.state.proto.mutator.extend.checkGuarantees"
-	ProtoStateMutatorExtendCheckSeals      = "common.state.proto.mutator.extend.checkSeals"
-	ProtoStateMutatorExtendCheckReceipts   = "common.state.proto.mutator.extend.checkReceipts"
-	ProtoStateMutatorExtendDBInsert        = "common.state.proto.mutator.extend.dbInsert"
+	ProtoStateMutatorExtend                SpanName = "common.state.proto.mutator.extend"
+	ProtoStateMutatorExtendCheckHeader     SpanName = "common.state.proto.mutator.extend.checkHeader"
+	ProtoStateMutatorExtendCheckGuarantees SpanName = "common.state.proto.mutator.extend.checkGuarantees"
+	ProtoStateMutatorExtendCheckSeals      SpanName = "common.state.proto.mutator.extend.checkSeals"
+	ProtoStateMutatorExtendCheckReceipts   SpanName = "common.state.proto.mutator.extend.checkReceipts"
+	ProtoStateMutatorExtendDBInsert        SpanName = "common.state.proto.mutator.extend.dbInsert"
```

Comment:
I don't know if this was a linter warning about best practices, but if it wasn't: for typed constants, I believe the type of the first constant is the type used for all the constants, so you really only need to specify it ones.

When constant type and value is not provided, then it gets its type and value from previous declaration. That's also how `iota` works.

Reply:
actually it turns out it is a lint error:
```
module/trace/constants.go:11:2: SA9004: only the first constant in this group has an explicit type (staticcheck)
        ProtoStateMutatorExtend                SpanName = "common.state.proto.mutator.extend"
```
I think what you describe only works with iota